osu.Game/Overlays/KeyBinding/KeyBindingsSubsection.cs

```diff
@@ -31,16 +32,21 @@ protected KeyBindingsSubsection(int? variant)
         }
 
         [BackgroundDependencyLoader]
-        private void load(KeyBindingStore store)
+        private void load(RealmContextFactory realmFactory)
         {
-            var bindings = store.Query(Ruleset?.ID, variant);
+            var rulesetId = Ruleset?.ID;
+
+            List<RealmKeyBinding> bindings;
+
+            using (var usage = realmFactory.GetForRead())
+                bindings = usage.Realm.All<RealmKeyBinding>().Where(b => b.RulesetID == rulesetId && b.Variant == variant).Detach();
```

Comment:
What's the difference between calling `Detach()` and `Freeze()` in this scenario? Is `Freeze()` more expensive because it creates a frozen realm?

Reply:
Freezing objects requires the source realm (the one which the freeze was run on) to be kept open, and disposed of once it's finished. Keeping frozen objects/realms open will keep that version of the realm on disk, which can lead to filesize bloat depending on how many follow-up changes are made while it's kept frozen.